app/src/main/java/com/kleshchin/danil/filemanager/MainActivity.java

```diff
@@ -0,0 +1,135 @@
+        setContentView(R.layout.activity_main);
+        listView = (ListView) findViewById(R.id.listView);
+        listItems = new ArrayList<>();
+        listFilesItems = new ArrayList<>();
+        listAdapter = new ListAdapter(getApplicationContext(), listItems);
+        initToolbar();
+        fillListView(new File(mainPath));
+        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                String path = ((TextView) findViewById(R.id.toolbar_title)).getText().toString() +
+                        "/" + listItems.get(i).fileName;
+                if (new File(path).isDirectory()) {
+                    fillListView(new File(path));
+                } else {
+                    Uri uri = Uri.fromFile(new File(path));
+                    Intent intent = new Intent(android.content.Intent.ACTION_VIEW);
+                    String mime = "*/*";
+                    MimeTypeMap mimeTypeMap = MimeTypeMap.getSingleton();
+                    if (mimeTypeMap.hasExtension(
+                            mimeTypeMap.getFileExtensionFromUrl(uri.toString())))
+                        mime = mimeTypeMap.getMimeTypeFromExtension(
+                                mimeTypeMap.getFileExtensionFromUrl(uri.toString()));
+                    intent.setDataAndType(uri, mime);
+                    startActivity(intent);
+                }
+            }
+        });
+    }
+
+    private void initToolbar() {
+        toolbar = (Toolbar) findViewById(R.id.main_toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayShowTitleEnabled(false);
+        toolbar.setLogo(new ColorDrawable(getResources().getColor(android.R.color.transparent)));
+        if (!((TextView) findViewById(R.id.toolbar_title)).getText().equals("") &&
+                !((TextView) findViewById(R.id.toolbar_title)).getText().equals(mainPath)) {
+            final Drawable upArrow = getResources().getDrawable(R.drawable.ic_arrow_back_black_36dp);
+            upArrow.setColorFilter(getResources().getColor(R.color.colorWhite), PorterDuff.Mode.SRC_ATOP);
+            toolbar.setLogo(upArrow);
+            setLogoOnClickListener(toolbar, new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    fillListView(new File(((TextView) findViewById(R.id.toolbar_title)).getText().
+                            toString().substring(0, ((TextView) findViewById(R.id.toolbar_title)).
+                            getText().toString().length() - 1 - currentDirectory.length() + 1)));
+                }
+            });
+        }
+    }
+
+    private void setLogoOnClickListener(Toolbar toolbar, View.OnClickListener listener) {
+        try {
+            Class<?> toolbarClass = Toolbar.class;
+            Field logoField = toolbarClass.getDeclaredField("mLogoView");
```

Comment:
Mmm, refleeeection. You don't need it in this case. Check the previous comment about the back button